packages/legend-data-cube/src/stores/core/DataCubeQuerySnapshotBuilder.ts

```diff
@@ -316,7 +344,12 @@ export function validateAndBuildQuerySnapshot(
   /** TODO: @datacube roundtrip */
 
   // --------------------------------- FILTER ---------------------------------
-  /** TODO: @datacube roundtrip */
+  if (funcMap.filter) {
+    data.filter = _buildFilterSnapshot(
+      _lambdaParam(funcMap.filter, 0).body[0]!,
+      filterOperations!,
```

Comment:
Avoid using the `non-null assertion operator` like this, this is pretty much a typing hack, and we strive to be ax explicit as we could when giving out proper message